### ISSUE TITLE
mup setup failed with custom SSL certificates

### ISSUE THREAD

**Issue by zhukovsp** [zodern/meteor-up#724]
Made first **mup setup** (mup 1.3.1) on clean Digital Ocean droplet (Ubuntu 16.04.3 x64)
Received error:
```
Failed to copy file  { Error: No such file
    at SFTPStream._transform (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/sftp.js:384:27)
    at SFTPStream.Transform._read (_stream_transform.js:167:10)
    at SFTPStream._read (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/sftp.js:170:15)
    at SFTPStream.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:333:12)
    at writeOrBuffer (_stream_writable.js:319:5)
    at SFTPStream.Writable.write (_stream_writable.js:246:11)
    at Channel.ondata (_stream_readable.js:555:20)
    at emitOne (events.js:96:13)
    at Channel.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Channel.Readable.push (_stream_readable.js:134:10)
    at SSH2Stream.<anonymous> (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/nodemiral/node_modules/ssh2/lib/Channel.js:140:15)
    at emitOne (events.js:96:13)
    at SSH2Stream.emit (events.js:188:7)
    at parsePacket (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/ssh.js:3163:10) code: 2, lang: '' }
Retrying in 3 seconds
Failed to copy file  { Error: No such file
    at SFTPStream._transform (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/sftp.js:384:27)
    at SFTPStream.Transform._read (_stream_transform.js:167:10)
    at SFTPStream._read (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/sftp.js:170:15)
    at SFTPStream.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:333:12)
    at writeOrBuffer (_stream_writable.js:319:5)
    at SFTPStream.Writable.write (_stream_writable.js:246:11)
    at Channel.ondata (_stream_readable.js:555:20)
    at emitOne (events.js:96:13)
    at Channel.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Channel.Readable.push (_stream_readable.js:134:10)
    at SSH2Stream.<anonymous> (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/nodemiral/node_modules/ssh2/lib/Channel.js:140:15)
    at emitOne (events.js:96:13)
    at SSH2Stream.emit (events.js:188:7)
    at parsePacket (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/ssh.js:3163:10) code: 2, lang: '' }
Retrying in 3 seconds
Failed to copy file  { Error: No such file
    at SFTPStream._transform (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/sftp.js:384:27)
    at SFTPStream.Transform._read (_stream_transform.js:167:10)
    at SFTPStream._read (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/sftp.js:170:15)
    at SFTPStream.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:333:12)
    at writeOrBuffer (_stream_writable.js:319:5)
    at SFTPStream.Writable.write (_stream_writable.js:246:11)
    at Channel.ondata (_stream_readable.js:555:20)
    at emitOne (events.js:96:13)
    at Channel.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Channel.Readable.push (_stream_readable.js:134:10)
    at SSH2Stream.<anonymous> (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/nodemiral/node_modules/ssh2/lib/Channel.js:140:15)
    at emitOne (events.js:96:13)
    at SSH2Stream.emit (events.js:188:7)
    at parsePacket (/Users/sergey/.nvm/versions/node/v6.9.1/lib/node_modules/mup/node_modules/ssh2-streams/lib/ssh.js:3163:10) code: 2, lang: '' }
Retrying in 3 seconds
```
My mup.js
```
module.exports = {
  servers: {
    one: {
      // TODO: set host address, username, and authentication method
      host: 'xxx.xx.xxx.xx',
      username: 'username',
      pem: '/Users/sergey/.ssh/id_rsa',
      // password: 'server-password'
      // or neither for authenticate from ssh-agent
    }
  },

  app: {
    // TODO: change app name and path
    name: 'myapp',
    path: '..',

    servers: {
      one: {},
    },

    buildOptions: {
      serverOnly: true,
    },

    env: {
      // TODO: Change to your app's url
      // If you are using ssl, it needs to start with https://
      ROOT_URL: 'https://myapp.tech',
      MONGO_URL: 'mongodb://localhost/meteor',
    },

    // ssl: { // (optional)
    //   // Enables let's encrypt (optional)
    //   autogenerate: {
    //     email: 'email.address@domain.com',
    //     // comma separated list of domains
    //     domains: 'website.com,www.website.com'
    //   }
    // },

    docker: {
      // change to 'kadirahq/meteord' if your app is using Meteor 1.3 or older
      image: 'abernix/meteord:base',
    },

    // Show progress bar while uploading bundle to server
    // You might need to disable it on CI servers
    enableUploadProgressBar: true
  },

  mongo: {
    version: '3.4.1',
    servers: {
      one: {}
    }
  },

  proxy: {
    domains:'myapp.tech',
    ssl: {
      crt: './bundle.crt', // this is a bundle of certificates
      key: './private.key', // this is the private key of the certificate
      forceSSL: true
    }
  },
};

```
then I comment block
```
// proxy: {
//    domains:'myapp.tech',
//    ssl: {
//      crt: './bundle.crt', // this is a bundle of certificates
//      key: './private.key', // this is the private key of the certificate
//      forceSSL: true
//    }
//  },
```

- run mup setup (success)
- uncomment proxy block
- mup setup again (success)
- mup deploy (success)

**Comment by alexgzhou** [comment]
@zodern When would this be fixed since the version is now 1.3.7, and the problem is still there?

**Comment by Floriferous** [comment]
Getting the same issue. For now I think the best option is to configure nginx manually.

**Comment by zodern** [comment]
This is fixed in mup 1.4.2.
